compiler/rustc_next_trait_solver/src/solve/assembly/mod.rs

```diff
@@ -146,31 +147,18 @@ where
         goal: Goal<I, Self>,
     ) -> Result<Candidate<I>, NoSolution>;
 
-    /// A type is `Sized` if its tail component is `Sized`.
+    /// A type is `Sized` if its tail component is `Sized`, a type is `MetaSized` if its tail
+    /// component is `MetaSized` and a type is `PointeeSized` if its tail component is
+    /// `PointeeSized`.
     ///
     /// These components are given by built-in rules from
-    /// [`structural_traits::instantiate_constituent_tys_for_sized_trait`].
-    fn consider_builtin_sized_candidate(
-        ecx: &mut EvalCtxt<'_, D>,
-        goal: Goal<I, Self>,
-    ) -> Result<Candidate<I>, NoSolution>;
-
-    /// A type is `MetaSized` if its tail component is `MetaSized`.
-    ///
-    /// These components are given by built-in rules from
-    /// [`structural_traits::instantiate_constituent_tys_for_sized_trait`].
-    fn consider_builtin_metasized_candidate(
-        ecx: &mut EvalCtxt<'_, D>,
-        goal: Goal<I, Self>,
-    ) -> Result<Candidate<I>, NoSolution>;
-
-    /// A type is `PointeeSized` if its tail component is `PointeeSized`.
-    ///
-    /// These components are given by built-in rules from
-    /// [`structural_traits::instantiate_constituent_tys_for_sized_trait`].
-    fn consider_builtin_pointeesized_candidate(
+    /// [`structural_traits::instantiate_constituent_tys_for_sized_trait`],
+    /// [`structural_traits::instantiate_constituent_tys_for_metasized_trait`] and
+    /// [`structural_traits::instantiate_constituent_tys_for_pointeesized_trait`].
```

Comment:
Maybe better to just refer to `instantiate_constituent_tys_for_sizedness_trait` only